pyproject.toml

```diff
@@ -37,7 +37,7 @@ dependencies = [
     "Automat >= 0.8.0",
     "hyperlink >= 17.1.1",
     "attrs >= 21.3.0",
-    "typing_extensions >= 3.10.0",
+    "typing_extensions >= 4.0.0",
```

Comment:
https://github.com/python/typing_extensions/blob/main/CHANGELOG.md#added-in-version-400 for `typing_extensions.Self`